jquery.flot.legend.js

```diff
@@ -71,80 +74,267 @@
+                legendEl.css('opacity', options.legend.backgroundOpacity);
             }
+        } else {
+            legendEl = $(html.join('')).appendTo(options.legend.container)[0];
+            options.legend.container.style.width = width + 'em';
+            options.legend.container.style.height = height + 'em';
         }
     }
 
+    // Define the shapes
+    var svgShapeDefs = `
+        <defs>
+            <symbol id="line" fill="none" viewBox="-5 -5 25 25">
+                <polyline points="0,15 5,5 10,10 15,0"/>
+            </symbol>
+
+            <symbol id="area" stroke-width="1" viewBox="-5 -5 25 25">
+                <polyline points="0,15 5,5 10,10 15,0, 15,15, 0,15"/>
+            </symbol>
+
+            <symbol id="bars" stroke-width="1" viewBox="-5 -5 25 25">
+                <polyline points="1.5,15.5 1.5,12.5, 4.5,12.5 4.5,15.5 6.5,15.5 6.5,3.5, 9.5,3.5 9.5,15.5 11.5,15.5 11.5,7.5 14.5,7.5 14.5,15.5 1.5,15.5"/>
+            </symbol>
+
+            <symbol id="circle" viewBox="-5 -5 25 25">
+                <circle cx="0" cy="15" r="2.5"/>
+                <circle cx="5" cy="5" r="2.5"/>
+                <circle cx="10" cy="10" r="2.5"/>
+                <circle cx="15" cy="0" r="2.5"/>
+            </symbol>
+
+            <symbol id="rectangle" viewBox="-5 -5 25 25">
+                <rect x="-2.1" y="12.9" width="4.2" height="4.2"/>
+                <rect x="2.9" y="2.9" width="4.2" height="4.2"/>
+                <rect x="7.9" y="7.9" width="4.2" height="4.2"/>
+                <rect x="12.9" y="-2.1" width="4.2" height="4.2"/>
+            </symbol>
+
+            <symbol id="diamond" viewBox="-5 -5 25 25">
+                <path d="M-3,15 L0,12 L3,15, L0,18 Z"/>
+                <path d="M2,5 L5,2 L8,5, L5,8 Z"/>
+                <path d="M7,10 L10,7 L13,10, L10,13 Z"/>
+                <path d="M12,0 L15,-3 L18,0, L15,3 Z"/>
+            </symbol>
+
+            <symbol id="cross" fill="none" viewBox="-5 -5 25 25">
+                <path d="M-2.1,12.9 L2.1,17.1, M2.1,12.9 L-2.1,17.1 Z"/>
+                <path d="M2.9,2.9 L7.1,7.1 M7.1,2.9 L2.9,7.1 Z"/>
+                <path d="M7.9,7.9 L12.1,12.1 M12.1,7.9 L7.9,12.1 Z"/>
+                <path d="M12.9,-2.1 L17.1,2.1 M17.1,-2.1 L12.9,2.1 Z"/>
+            </symbol>
+
+            <symbol id="plus" fill="none" viewBox="-5 -5 25 25">
+                <path d="M0,12 L0,18, M-3,15 L3,15 Z"/>
+                <path d="M5,2 L5,8 M2,5 L8,5 Z"/>
+                <path d="M10,7 L10,13 M7,10 L13,10 Z"/>
+                <path d="M15,-3 L15,3 M12,0 L18,0 Z"/>
+            </symbol>
+        </defs>`;
+
+    function getEntryHtml(shape) {
+        var html = '',
+            name = shape.name,
+            x = shape.xPos,
+            y = shape.yPos,
+            fill = shape.fillColor,
+            stroke = shape.strokeColor,
+            width = shape.strokeWidth;
+        switch (name) {
+            case "circle":
```

Comment:
Let's stick to ' instead of " for the first level of nesting, and " for the second level.